boards/adafruit-pybadge/Makefile.dep

```diff
@@ -3,6 +3,11 @@ ifneq (,$(filter saul_default,$(USEMODULE)))
 endif
 
 ifneq (,$(filter disp_dev,$(USEMODULE)))
+  USEMODULE += st77xx
+endif
+
+ifneq (,$(filter st77xx,$(USEMODULE)))
+  # indicate that a ST7735 is used
   USEMODULE += st7735
 endif
```

Comment:
I think the entry point for makefile.dep is the specific model, thus only using the default `disp_dev` to select `st7735` would be fine:
```suggestion
ifneq (,$(filter disp_dev,$(USEMODULE)))
    USEMODULE += st7735
endif
```

Reply:
@MrKevinWeiss First, thanks for your review. Unfortunalty, the PR depends on PR #19825 and includes it for now, that's why it is labeled `Stat: Waiting for another PR`. This change is part of PR #19825 where @aabadie already suggested the same change.

In comment https://github.com/RIOT-OS/RIOT/pull/19825#discussion_r1264386938 I explained why I used `st77xx` by intention here. For short:

- Kconfig can select only the driver in `app.config.test` of `tests/drivers/st77xx`, the board then selects the right variant based on `HAVE_ST77*`. Otherwise you would need a test app for each driver variant (just code duplication).
- Enabling the driver module `st77xx` here instead of the variant avoids to blow up `drivers/Makefile.dep` by driver dependency details.
- It seems to be more logical from my point of view that an app just selects the driver and the boards says what variant it uses.

`si114x` is a bad example since it hard codes `si1145` in `tests/drivers/si114x` (in the `Makefile` as well as the `app.config.test`). Therefore, other variants aren't covered by CI compilation :worried: This is exactly what I tried to prevent.

Even more, if a board has a certain display, the right `st77*` variant should be enabled by default, this is what I would expect. It makes no sense to compile a board with `st7735` if it has a `st7796` display. IMHO the board developer should be responsible for selecting the correct `st77*` variant and providing the correct `HAVE_ST*` function just as he is responsible for defining the correct hardware parameters for the driver in `board.h`. Maybe I misunderstood the design goal of Kconfig til now, but I thought the board should select by default what it supports based on `HAS_*` and `HAVE_*` and denabled modules.

Reply:
This was part of PR #19825 ans has been resolved in that PR.